src/dc_receive_imf.rs

```diff
@@ -2892,6 +2902,65 @@ mod tests {
         assert_eq!(msgs.len(), 2);
     }
 
+    #[async_std::test]
+    async fn test_majordomo_mailing_list() {
+        let t = TestContext::new_alice().await;
+        t.ctx
+            .set_config(Config::ShowEmails, Some("2"))
+            .await
+            .unwrap();
+
+        // test mailing lists not having a `ListId:`-header
+        dc_receive_imf(
+            &t.ctx,
+            b"From: Foo Bar <foo@bar.org>\n\
+    To: deltachat/deltachat-core-rust <deltachat-core-rust@noreply.github.com>\n\
+    Subject: [ola] just a subject\n\
+    Message-ID: <3333@example.org>\n\
+    Sender: My list <mylist@bar.org>\n\
+    Precedence: list\n\
+    Date: Sun, 22 Mar 2020 22:37:57 +0000\n\
+    \n\
+    hello\n",
+            "INBOX",
+            1,
+            false,
+        )
+        .await
+        .unwrap();
+        let msg = t.get_last_msg().await;
+        let chat_id =
+            message::decide_on_contact_request(&t, msg.id, ContactRequestDecision::StartChat)
+                .await
+                .unwrap();
+        let chat = Chat::load_from_db(&t, chat_id).await.unwrap();
+        assert_eq!(chat.typ, Chattype::Mailinglist);
+        assert_eq!(chat.grpid, "mylist@bar.org");
+        assert_eq!(chat.name, "ola");
+
+        // receive another message with no sender name but the same address,
+        // make sure this lands in the same chat
+        dc_receive_imf(
+            &t.ctx,
+            b"From: Nu Bar <nu@bar.org>\n\
+    To: deltachat/deltachat-core-rust <deltachat-core-rust@noreply.github.com>\n\
+    Subject: [ola] Re: just a subject\n\
+    Message-ID: <4444@example.org>\n\
+    Sender: mylist@bar.org\n\
+    Precedence: list\n\
+    Date: Sun, 22 Mar 2020 23:37:57 +0000\n\
+    \n\
+    hello\n",
+            "INBOX",
+            1,
+            false,
+        )
+        .await
+        .unwrap();
+        let msgs = chat::get_chat_msgs(&t, chat.id, 0, None).await;
+        assert_eq!(msgs.len(), 2);
```

Comment:
If we want to make sure that no new chat is created, we may want to add another `assert_eq!(msgs.len(), 2);` before receiving the second message. Or maybe even check that the chat ids match.

Reply:
+1